src/main/java/sigma/Sigma.java

```diff
@@ -0,0 +1,422 @@
+        LocalDateTime from = LocalDateTime.parse(timeParts[0], formatter);
+        LocalDateTime to = LocalDateTime.parse(timeParts[1], formatter);
+        if (description.isEmpty()) {
+            throw new SigmaMissingArgException(CommandType.EVENT);
+        }
+        if (to.isBefore(from)) {
+            throw new SigmaInvalidDateRangeException();
+        }
+        return new Event(description, from, to);
+    }
+
+    /**
+     * Handles the "delete" command, removing a task from the list.
+     *
+     * @param userInput The task number to delete.
+     * @throws SigmaException If the task number is invalid.
+     */
+    private String handleDeleteCommand(String userInput) throws SigmaException {
+        try {
+            int taskNumber = Integer.parseInt(userInput);
+            if (taskNumber < 1 || taskNumber > taskList.getSize() || taskList.getSize() == 0) {
+                throw new SigmaInvalidTaskException(taskNumber);
+            }
+            assert taskList.getSize() != 0;
+            String s = this.ui.showDeletedTask(taskList.getTask(taskNumber), taskList.getSize() - 1);
+            taskList.deleteTask(taskNumber);
+            assert taskList.getSize() == (taskList.getSize() + 1) - 1;
+            return s;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new SigmaInvalidArgException(CommandType.DELETE);
+        } catch (NumberFormatException e) {
+            throw new SigmaNaNException();
+        }
+    }
+
+    /**
+     * Handles the "find" command, searching for tasks containing the keyword.
+     *
+     * @param userInput The keyword to search for in tasks.
+     * @throws SigmaException If the keyword is missing or invalid.
+     */
+    public String handleFindCommand(String userInput) throws SigmaException {
+        if (taskList.getSize() == 0) {
+            throw new SigmaEmptyTaskListException();
+        }
+
+        if (userInput.trim().split(" ").length > 1) {
+            throw new SigmaInvalidArgException(CommandType.FIND);
+        }
+
+        return this.ui.showSearchedTasks(this.taskList.search(userInput));
+    }
+
+    /**
+     * Updates an existing task in the task list based on the user's input.
+     * <p>
+     * The format for the user input is "{taskNumber} {updateDetails}". For {@code Deadline} tasks,
+     * {@code updateDetails} can be "description /by dd/MM/yy HH:mm". For {@code Event} tasks,
+     * it can be "description /from dd/MM/yy HH:mm /to dd/MM/yy HH:mm".
+     * <p>
+     * If the task number is invalid or if the update details are missing or incorrectly formatted,
+     * appropriate exceptions will be thrown.
+     *
+     * @param userInput The input command with task number and update details.
+     * @return A confirmation message of the update.
+     * @throws SigmaException If the task number is invalid, or if the update details are incorrect.
+     */
+    public String handleUpdateCommand(String userInput) throws SigmaException {
```

Comment:
Very cool that you introduced an update task function 🦑🦑🦑🦑
Suggestion: maybe you can also let users know when the task was last updated 😁